auth_backends/suomifi.py

```diff
@@ -281,8 +287,16 @@ def extra_data(self, user, uid, response, details=None, *args, **kwargs):
         Extends the base class method by including session index to extra_data."""
         data = super().extra_data(user, uid, response, details=details, *args, **kwargs)
         data['session_index'] = response.get('session_index')
+        data['suomifi_attributes'] = self._extract_suomifi_attributes(response)
         return data
 
+    def _extract_suomifi_attributes(self, response):
+        attributes = {}
+        for attribute in SuomiFiUserAttribute.objects.all():
+            if attribute.uri in response.get('attributes'):
+                attributes[attribute.friendly_name] = response.get('attributes')[attribute.uri][0]
```

Comment:
The attribute value is a list because the original XML element can contain many children. Is it specified somewhere that there will only be one value?

Reply:
Although the SAML specification allows for multiple values per attribute the Suomi.fi documentation describing the attributes implies only a single value for each attribute.